core/tests/protractor/accessibility.js

```diff
@@ -41,6 +120,164 @@ describe('screenreader and keyboard user accessibility features', function() {
+      var skipButton = element(by.css('.protractor-test-skip-link'));
+      expect(await skipButton.getAttribute('id')).toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+
+      // Should move the focus away from the skip to main content button.
+      await browser.actions().sendKeys(protractor.Key.TAB).perform();
+      await browser.actions().sendKeys(protractor.Key.TAB).perform();
+      expect(await skipButton.getAttribute('id')).not.toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+
+      // Should move the focus to the skip to main content button.
+      await browser.actions().sendKeys('s').perform();
+      expect(await skipButton.getAttribute('id')).toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+    });
+
+  it('should move focus to the search bar in the library page',
+    async function() {
+      // Should move the focus to the search bar.
+      await libraryPage.get();
+      await browser.actions().sendKeys('/').perform();
+      var searchBar = element(by.css('.protractor-test-search-input'));
+      expect(await searchBar.getAttribute('id')).toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+
+      // Should move the focus away from the search bar.
+      await browser.actions().sendKeys(protractor.Key.TAB).perform();
+      await browser.actions().sendKeys(protractor.Key.TAB).perform();
+      expect(await searchBar.getAttribute('id')).not.toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+
+      // Should move the focus to the search bar.
+      await browser.actions().sendKeys('/').perform();
+      expect(await searchBar.getAttribute('id')).toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+    });
+
+  it('should move focus to the category bar in library page',
+    async function() {
+      // Should move the focus to the category bar.
+      await libraryPage.get();
+      await browser.actions().sendKeys('c').perform();
+      var categoryBar = element(by.css(
+        '.protractor-test-search-bar-dropdown-toggle'));
+      expect(await categoryBar.getAttribute('id')).toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+
+      // Should move the focus away from the category bar.
+      await browser.actions().sendKeys(protractor.Key.TAB).perform();
+      await browser.actions().sendKeys(protractor.Key.TAB).perform();
+      expect(await categoryBar.getAttribute('id')).not.toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+
+      // Should move the focus to the category bar.
+      await browser.actions().sendKeys('c').perform();
```

Comment:
These tests often seem to follow the same basic pattern: send a key and then check that the focus has moved to or from an element. This seems like it could be made into a function you reuse across these tests?